### PR TITLE
Removing iconUrl To Make Debugging Easy

### PR DESCRIPTION
**Pull request by SIMRAN1**
Thanks for opening a Pull Request (PR), here's a few guidelines as to what we need in your PR before we review it.
Fixes#2595
Please delete anything that isn't relevant to your patch.

- [x] This PR relates to an existing open issue and there are no existing PRs open for the same issue.
- [x] Add `Fixes #ISSUENUM` at the top of your PR.
- [x] Add a description of the the changes introduced in this PR.
- [ ] The change has been successfully run locally.
- [ ] Add tests to cover the changes added in this PR.
- [ ] Add before and after screenshots (Only for changes that impact the UI).

Once you have met the above requirements please replace this section with a `Fixes #ISSUENUM` linking to the issue fixed by this PR along with an explanation of the changes. Thanks for your contribution!

### CODE REVIEW

**Comment by SIMRAN1** [comment]
@willdurand please review!

File: src/core/reducers/addons.js [hunk 1]
```diff
@@ -194,7 +194,7 @@ export function createInternalAddon(apiAddon: ExternalAddonType): AddonType {
     // TODO: remove this if possible. This is used by core/installAddon
     // and DiscoPane components which do camel case conversions for
     // some historic reason.
```
Comment (championshuttler) [inline src/core/reducers/addons.js]: @SIMRAN1, thanks for the PR,  I think you should remove the comments too :)

File: src/core/reducers/addons.js [hunk 2]
```diff
@@ -194,7 +194,7 @@ export function createInternalAddon(apiAddon: ExternalAddonType): AddonType {
...
-    iconUrl: apiAddon.icon_url,
+
```
Comment (championshuttler) [inline src/core/reducers/addons.js]: Also please check https://travis-ci.org/mozilla/addons-frontend/jobs/420119929#L515 , you have trailing space here.
Please make sure to run `yarn prettier-dev` before commit

Comment (willdurand) [inline src/core/reducers/addons.js]: I don't think this is enough. The `iconUrl` property is used in the code as described in the code comment above.

File: src/core/reducers/addons.js (hunk 1 above)
Comment (SIMRAN1) [reply src/core/reducers/addons.js]: Thanks @championshuttler !

File: src/core/reducers/addons.js (hunk 2 above)
Comment (SIMRAN1) [reply src/core/reducers/addons.js]: okk @championshuttler !

Comment (SIMRAN1) [reply src/core/reducers/addons.js]: okk @willdurand !

Comment (SIMRAN1) [reply src/core/reducers/addons.js]: @willdurand I have a question. i have to refractor iconUrl to icon_url or remove it!

Comment (willdurand) [reply src/core/reducers/addons.js]: it is about removing iconUrl but this property is unfortunately used in some places.

File: src/disco/components/Addon/index.js [hunk 3]
```diff
@@ -99,17 +98,6 @@ export class AddonBase extends React.Component {
     ) : null;
   }
 
-  getLogo() {
```
Comment (willdurand) [inline src/disco/components/Addon/index.js]: I don't think this code should be removed.

File: src/disco/components/Addon/index.js [hunk 4]
```diff
@@ -57,7 +57,6 @@ export class AddonBase extends React.Component {
     hasAddonManager: PropTypes.bool.isRequired,
     heading: PropTypes.string.isRequired,
     i18n: PropTypes.object.isRequired,
-    iconUrl: PropTypes.string,
```
Comment (willdurand) [inline src/disco/components/Addon/index.js]: You can leave this prop actually, but you have to assign it in `mapStateToProps()` I think:

```js
export function mapStateToProps(state, ownProps) {
  // `ownProps.guid` is already "normalized" with `getGuid()` in the
  // `DiscoPane` container component.
  const installation = state.installations[ownProps.guid] || {};
  const addon = getAddonByGUID(state, ownProps.guid);

  return {
    addon,
    ...addon,
    iconUrl: addon ? addon.icon_url : null, // <----
    ...installation,
    status: installation.status || UNKNOWN,
    clientApp: state.api.clientApp,
    // This is required by the `withInstallHelpers()` HOC, apparently...
    platformFiles: addon ? addon.platformFiles : {},
    userAgentInfo: state.api.userAgentInfo,
  };
}
```

File: src/disco/components/Addon/index.js [hunk 5]
```diff
@@ -275,7 +263,6 @@ export class AddonBase extends React.Component {
       /* eslint-disable jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/click-events-have-key-events */
       <div className={addonClasses}>
         {this.getThemeImage()}
-        {this.getLogo()}
```
Comment (willdurand) [inline src/disco/components/Addon/index.js]: You should keep this code too.

File: src/core/types/addons.js [hunk 6]
```diff
@@ -82,7 +82,7 @@ export type ThemeData = {|
   footerURL?: string,
   header?: string,
   headerURL?: string,
-  iconURL?: string,
+  icon_url?: string,
```
Comment (willdurand) [inline src/core/types/addons.js]: I think you can remove this prop.

File: src/core/types/addons.js [hunk 7]
```diff
@@ -158,7 +158,7 @@ export type AddonType = {|
   ...ExternalAddonType,
   ...ThemeData,
   // Here are some custom properties for our internal representation.
-  iconUrl?: string,
+  icon_url?: string,
```
Comment (willdurand) [inline src/core/types/addons.js]: You can probably remove this prop too.

**Comment by willdurand** [comment]
Hi @SIMRAN1, I am going to close this PR as it has been updated and it now has conflicts. If you want to continue working on the issue, please open another PR! (no problem with opening new PRs)

**Pull request closed without merging**